pkg/build/controller/controller.go

```diff
@@ -89,9 +89,12 @@ func (bc *BuildController) nextBuildStatus(build *buildapi.Build) error {
 		repo, err := bc.ImageStreamClient.GetImageStream(namespace, ref.Name)
 		if err != nil {
 			if errors.IsNotFound(err) {
-				return fmt.Errorf("the referenced output image repository %s/%s does not exist", namespace, ref.Name)
+				return fmt.Errorf("the referenced output image stream %s/%s does not exist", namespace, ref.Name)
 			}
-			return fmt.Errorf("the referenced output repo %s/%s could not be found by %s/%s: %v", namespace, ref.Name, build.Namespace, build.Name, err)
+			return fmt.Errorf("the referenced output image stream %s/%s could not be found by build %s/%s: %v", namespace, ref.Name, build.Namespace, build.Name, err)
+		}
+		if len(repo.Status.DockerImageRepository) == 0 {
+			return fmt.Errorf("the integrated Docker registry has not be configured and so the image stream %s/%s cannot be used as the output for build %s/%s", namespace, ref.Name, build.Namespace, build.Name)
```

Comment:
the more i think about it the more this error bothers me.  it should really be saying "the image stream does not have a docker registry associated with it, and therefore it cannot be used.  This could be because the integrated docker registry has not been configured"  (it could also be because you meant to reference an external registry, and forgot to do so).

Admittedly most deployments will have the integrated docker registry, or intend to use it, so the second case is unlikely to occur, but it is possible.